tools/sctrace/src/main.rs

```diff
@@ -0,0 +1,262 @@
+    ///
+    /// * `syscall` - The parsed syscall to validate and potentially output
+    ///
+    /// # Output Behavior
+    ///
+    /// | Mode   | Supported Syscall | Unsupported Syscall |
+    /// |--------|-------------------|---------------------|
+    /// | Normal | Print line        | Print with red "(unsupported)" |
+    /// | Quiet  | No output         | Print with "unsupported:" prefix |
+    fn handle(&self, syscall: &Syscall) {
+        match self.matcher.match_syscall(syscall) {
+            Some(_) => {
+                // Syscall is supported, print if not in quiet mode
+                if !self.params.quiet() {
+                    println!("{}", syscall.original_line());
+                }
+            }
+            None => self.print_unsupported_syscall(syscall),
+        }
+    }
+
+    /// Prints unsupported syscall with appropriate formatting based on quiet mode.
+    ///
+    /// The output format varies depending on the quiet mode setting:
+    ///
+    /// - **Quiet mode**: Prefixes the line with `"unsupported: "`
+    /// - **Normal mode**: Appends a red-colored `"(unsupported)"` suffix using ANSI escape codes
+    ///
+    /// # Arguments
+    ///
+    /// * `syscall` - The unsupported syscall to print
+    fn print_unsupported_syscall(&self, syscall: &Syscall) {
+        if self.params.quiet() {
+            println!("unsupported: {}", syscall.original_line());
+        } else {
+            println!("{} \x1b[31m(unsupported)\x1b[0m", syscall.original_line());
+        }
+    }
+}
+
+/// Main entry point for the sctrace tool.
+///
+/// This function orchestrates the entire syscall tracing and validation process:
+///
+/// # Workflow
+///
+/// 1. **Parse Arguments**: Validates and extracts command-line parameters
+/// 2. **Load Patterns**: Reads and parses the SCML pattern file
+/// 3. **Initialize Tracer**: Creates an appropriate trace iterator (online/offline)
+/// 4. **Process Syscalls**: Iterates through trace lines, parsing and validating each syscall
+/// 5. **Report Results**: Outputs supported/unsupported syscalls with appropriate formatting
+///
+/// # Error Handling
+///
+/// The function exits with code `1` if any of the following errors occur:
+/// - Invalid command-line arguments
+/// - SCML file not found or malformed
+/// - Failed to start tracing (online mode)
+/// - Input log file not found (offline mode)
+///
+/// Parse errors for individual syscall lines are reported to stderr but don't terminate execution.
+///
+/// # Exit Codes
+///
+/// - `0` - All operations completed successfully
+/// - `1` - Fatal error during initialization or setup
+///
+/// # Examples
+///
+/// Running the tool:
+///
+/// ```bash
+/// # Online mode - trace ls command
+/// sctrace patterns.scml -- /bin/ls -la
+///
+/// # Offline mode - analyze existing log
+/// sctrace patterns.scml --input trace.log
+///
+/// # Quiet mode - only show unsupported
+/// sctrace --quiet patterns.scml -- /usr/bin/gcc test.c
+/// ```
+fn main() {
```

Comment:
The target of the integration test is the high-level API provided by `lib.rs`, yet `main.rs` does not use those APIs at all. Instead, it chooses to implement the CLI tool using lower-level APIs from the ground up. This situation is obviously undesirable.